eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,32 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.anySatisfy(this.adapted, each -> {
+            Iterable<V> iterable = this.function.valueOf(each);
+            if (iterable == null)
+            {
+                throw new NullPointerException();
+            }
+            return Iterate.anySatisfy(iterable, each1 -> {
+                if (predicate.accept(each1))
+                {
+                    if (each == null)
+                    {
+                        throw new NullPointerException();
+                    }
+                    result[0] = each1;
```

Comment:
Good catch! 

I actually meant to check `each1` here. I reviewed my test case and find one corner case is missing there. I added missing test in `FlatCollectIterableTestNoIteratorTest`, moved `each` check out of lambda and put `each1` check here.